src/dataset.js

```diff
@@ -121,6 +156,107 @@ export default class Dataset {
     this._observe(this.state = state);
   }
 
+  // Applies the filter to all possible Resolved Pages
+  refilter(){
+    let readOffset = this.state.readOffset;
+    let state = this.state.update((next)=> {
+      next.readOffset = readOffset;
+      let pages  = next.pages;
+
+      let minLoadPage = Math.floor((readOffset  - next.loadHorizon) / next.pageSize);
+      let maxLoadPage = Math.ceil((readOffset  + next.loadHorizon) / next.pageSize);
+      let minUnloadPage = Math.floor((readOffset - next.unloadHorizon) / next.pageSize);
+      let maxUnloadPage = Math.ceil((readOffset  + next.unloadHorizon) / next.pageSize);
+
+      var minLoadHorizon = Math.max(minLoadPage, 0);
+      var maxLoadHorizon = Math.min(next.stats.totalPages || Infinity, maxLoadPage);
+      var minUnloadHorizon = Math.max(minUnloadPage, 0);
+      var maxUnloadHorizon = Math.min(next.stats.totalPages || Infinity, maxUnloadPage, next.pages.length);
+
+      // Filter Pages between current Horizons
+      let currentMinHorizon = Math.min(minUnloadHorizon, minLoadHorizon);
+      let currentMaxHorizon = Math.max(maxUnloadHorizon, maxLoadHorizon);
+      for (var i = currentMinHorizon; i < currentMaxHorizon; i += 1) {
+        this._filterPage(pages, i);
+      }
+
+      // Update the total number of records after applying filters
+      this._adjustTotalRecords(next);
+    });
+    this._observe(this.state = state);
+  }
+
+  // Unload all pages, and refetch from the readOffset (default: 0)
+  // Reload does `unfetch` all unloaded pages
+  reload(readOffset){
```

Comment:
I agree that the default should be the current readOffset.

However, let's say we want to reload our dataset from the beginning. Let's say the current readOffset is 100, and we wish to set it to 0. Reload will then `unfetch` all pages and start at the beginning.